tests/WebSharper.Html5.Tests/JSBindings.fs

```diff
@@ -0,0 +1,252 @@
+﻿// $begin{copyright}
+//
+// This file is part of WebSharper
+//
+// Copyright (c) 2008-2016 IntelliFactory
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.  See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// $end{copyright}
+
+module WebSharper.Html5.Tests.JSBindings
+
+open WebSharper
+open WebSharper.JavaScript
+open WebSharper.JQuery
+open WebSharper.Testing
+
+
+type Browser =
+    | IE of string
+    | Firefox of string
+    | Edge of string
+    | Chrome of string
+    | Safari of string
+    | Opera of string
+    | Other of string
+
+[<JavaScript>]
+let BrowserVersion() =
+    let ua = JS.Window.Navigator?userAgent
+    let ua = ua.ToString()
+    let regexstr = "(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)"
+    let regexp = new RegExp(regexstr, "i")
+    let m = regexp.Exec(ua)
+    let m1, m2 =
+        if Array.length m >= 3
+            then m.[1], m.[2]
+            else if Array.length m >= 2
+                then m.[1], ""
+                else "", ""
+    if (new RegExp("trident", "i")).Test(m1) then
+        let version = "\\brv[ :]+(\d+)"
+        let iregexp = new RegExp(version, "g")
+        let res = iregexp.Exec(ua)
+        let ver = if Array.length res >= 2 then res.[1] else ""
+        (Browser.IE ver)
+    else if (m1 = "Chrome") then
+        let iregexp = new RegExp("\\b(OPR|Edge)\/(\d+)", "")
+        let im = iregexp.Exec(ua)
+        match im with
+        | null -> 
+            (Browser.Chrome m2)
+        | im ->
+            let res = im.[1..]
+            let br = res.[1]
+            match br with
+            | "OPR" -> (Browser.Opera res.[2])
+            | "Edge" -> (Browser.Edge res.[2])
+            | _ -> (Browser.Other ua)
+    else if (m1 = "Firefox") then
+        (Browser.Firefox m2)
+    else
+        (Browser.Other ua)
+[<JavaScript>]
+let isNotIE () =

```

Comment:
It's generally a bad idea to make a function `isNotX`, you should rather have `isIE` and call it with `not (isIE x)`.